src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -163,14 +168,62 @@ public KeyManager[] getKeyManagers(URI privateKeyUri, URI certificateUri)
         return provider.getKeyManagers(privateKeyUri, certificateUri);
     }
 
+    /**
+     * Get certificate chain using private and certificate URIs.
+     *
+     * @param privateKeyUri  private key URI
+     * @param certificateUri certificate URI
+     * @return X509Certificate list that corresponds to the given private key and certificate URIs
+     * @throws ServiceUnavailableException if crypto key provider service is unavailable
+     * @throws KeyLoadingException         if crypto key provider service fails to load key
+     * @throws CertificateException if a valid certificate is not found at the URI specified
+     * @throws IOException         if reading file at the specified URI fails
+     */
+    public List<X509Certificate> getCertificateChain(URI privateKeyUri, URI certificateUri)
+            throws ServiceUnavailableException, KeyLoadingException, IOException, CertificateException {
+        logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                .log("Getting the certificate chain using private key and certificate URIs");
+        CryptoKeySpi provider = selectCryptoKeyProvider(privateKeyUri);
+        try {
+            return provider.getCertificateChain(privateKeyUri, certificateUri);
+        } catch (UnsupportedOperationException e) {
+            logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                    .log("Could not find the certificate using the provider implementation."
+                            + " Getting it from the key manager.");
+            KeyManager[] km = provider.getKeyManagers(privateKeyUri, certificateUri);
+            if (km.length != 1 || !(km[0] instanceof X509KeyManager)) {
```

Comment:
Can we guarantee that the X509KeyManager is always going to be returned in position 0?. What if for some reason the API changes and it gets returned in position 1 and the length is greater that 1?. This is forcing the client to have implicit understanding of how the provider stores the managers. I believe we should just tell the provider to give us the `x509KeyManager` like

```
X509KeyManager x509KeyManager = provider.getKeyManager(pkUri, certUri, X509)

if (x509KeyManager == null) {
    return Collections.emptyList()
}
```

Reply:
In both file and pkcs providers, we create a keystore with only one key along with its **x509** type certs and `init` the keymanager instance with it. So, we'll only have one key manager and that will be of type `X509KeyManager`. Otherwise, it is neither guaranteed that it will be of size 1 not x509 can be found at index 0.

Reply:
Got it, but the caller basically needs to have all that information an do this type of checks every time they want to get the `X509KeyManager`.  The disadvantages of this is that

* the caller is aware of the data structure we are using
* It makes things harder to change, if we want to change how the provider stores the managers we then have to make sure we change all the callers.

Reply:
`!A || !B` logic that you want to evaluate to false? You're making me recruit way too many brain cells here. Please invert this condition logic and flip the if/else.

`km.length == 1 && km[0] instanceof X509KeyManager`

Reply:
Had to code this way initially to avoid PMD violation for nested if statements. Updated it.